cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -600,6 +606,71 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, specification);
   }
 
+  /**
+   * Get schedules containing {@link ProgramStatusTrigger} filtered by triggering program
+   * and triggering program statuses.
+   *
+   * @param format format of the returned schedules. Use "detail" to return {@link ScheduleDetail}
+   *               or "spec" to return {@link ScheduleSpecification}
+   * @param triggerType trigger type of returned schedules. If not specify, all schedules
+   *                    are returned regardless of trigger type
+   * @param triggerNamespaceId namespace of the triggering program in {@link ProgramStatusTrigger}
+   * @param triggerAppName application name of the triggering program in {@link ProgramStatusTrigger}
+   * @param triggerAppVersion application version of the triggering program in {@link ProgramStatusTrigger}
+   * @param triggerProgramType program type of the triggering program in {@link ProgramStatusTrigger}
+   * @param triggerProgramName program name of the triggering program in {@link ProgramStatusTrigger}
+   * @param triggerProgramStatuses comma separated {@link ProgramStatus} in {@link ProgramStatusTrigger}.
+   *                               Schedules with {@link ProgramStatusTrigger} triggered by none of the
+   *                               {@link ProgramStatus} in triggerProgramStatuses will be filtered out.
+   */
+  @GET
+  @Path("schedules/trigger-type/program-status")
+  public void getProgramStatusSchedules(HttpRequest request, HttpResponder responder,
+                                        @QueryParam("format") String format,
+                                        @QueryParam("trigger-type") String triggerType,
+                                        @QueryParam("trigger-namespace-id") String triggerNamespaceId,
+                                        @QueryParam("trigger-app-name") String triggerAppName,
+                                        @QueryParam("trigger-app-version") String triggerAppVersion,
+                                        @QueryParam("trigger-program-type") String triggerProgramType,
+                                        @QueryParam("trigger-program-name") String triggerProgramName,
+                                        @QueryParam("trigger-program-statuses") String triggerProgramStatuses)
+    throws BadRequestException {
+    boolean asScheduleSpec = returnScheduleAsSpec(format);
+    if (triggerNamespaceId == null && triggerAppName == null && triggerAppVersion == null
+      && triggerProgramType == null && triggerProgramName == null) {
```

Comment:
that's also intellij formatting. two spaces to indicate String concatenation with the previous line